backend/src/directors/report_director.py

```diff
@@ -19,13 +25,24 @@ async def create_report_from_file(upload: UploadFile) -> ReportResponse:
     if file_size > MAX_FILE_SIZE:
         raise HTTPException(status_code=413, detail=f"File upload must be less than {MAX_FILE_SIZE} bytes")
 
-    file = LLMFile(filename=upload.filename, file=file_stream)
     file_id = str(uuid.uuid4())
+    file = LLMFile(id=file_id, filename=upload.filename, file=file_stream)
+
+    session_file = FileUpload(
+        id=file_id,
+        filename=file.filename,
+        upload_id=None,
+        content=None
+    )
+
+    update_session_file_uploads(session_file)
 
     report_agent = get_report_agent()
 
     company_name = await report_agent.get_company_name(file)
 
+    update_session_file_upload_with_company(session_file, company_name)
+
```

Comment:
I think we are just storing the filename and company name in redis so we can link them together later? Is it possible to extract this to a single method somewhere in the report director so that it would look something like:

```suggestion
    file = LLMFile(filename=upload.filename, file=file_stream)
    
    report_agent = get_report_agent()

    company_name = await report_agent.get_company_name(file)

    add_report_file_to_session(file.filename, company_name)
    
    .....
def add_report_file_to_session(file_id: str, filename: str, company_name: str) -> None:    
    session_file = FileUpload(
        filename=filename,
        upload_id=None,
        content=None
    )

    update_session_file_uploads(session_file)

    update_session_file_upload_with_company(session_file, company_name)  # we could even condense these last two lines into a single method as well, possibly
```